lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -140,3 +150,70 @@ defaultRetryPolicy =
     limitRetriesByCumulativeDelay (60 * second) (exponentialBackoff 10000)
   where
     second = 1000*1000
+
+-- | Subscribe to a blockchain and get called with new block (in order)!
+follow
+    :: NetworkLayer IO tx block
+    -- ^ The @NetworkLayer@ used to poll for new blocks.
+    -> Trace IO Text
+    -> BlockHeader
+    -- ^ The local tip to start at. Blocks /after/ the tip will be yielded.
+    -> (NE.NonEmpty block -> BlockHeader -> ExceptT e IO ())
+     -- ^ Callback with blocks and the new tip. @follow@ stops polling and
+     -- terminates if the callback errors.
+    -> (block -> BlockHeader)
+    -> IO ()
+follow nl tr start yield header = do
+    liftIO $ runExceptT (networkTip nl) >>= \case
+        Right nwTip ->
+            restoreStep (start, nwTip)
+        Left e -> do
+            logError tr $ "Failed to get network tip: " +|| e ||+ ""
+            restoreSleep start
+  where
+    restoreStep
+        :: (BlockHeader, BlockHeader)
+        -> IO ()
+    restoreStep (localTip, nodeTip) = do
+        runExceptT (nextBlocks nl localTip) >>= \case
+            Left e -> do
+                logError tr $ "Failed to get next blocks: " +|| e ||+ "."
+                restoreSleep localTip
+            Right [] -> do
+                logDebug tr "caught up with the node."
+                restoreSleep localTip
+            Right (blockFirst : blocksRest) -> do
+                let blocks = blockFirst NE.:| blocksRest
+
+                -- Get the tip of the last block, /before/ yielding the blocks.
+                -- This could potentially be avoided.
```

Comment:
I was a bit confused how everything should work at first, and thought the helper would know about, and manage the state to some extent. We *could* still return the new tip in the callback…

But removing the comment, probably.